### ISSUE TITLE
[SUGGESTION] For all builds, have this mod turn itself off on server-ends

### ISSUE THREAD

**Issue by Sunconure11** [Numelon-Softworks/ThreatenGL#8]
Just so you can have less dumb crash reports.

**Comment by Richy-Z** [comment]
Hello, thank you for this suggestion!
Unfortunately I wont implement this because some true sigmas told me that they want 9932497 fps in the Minecraft Server Profiler. 😁
![skibidi-fps-in-profiler](https://github.com/Numelon-Softworks/ThreatenGL/assets/64844585/33d74f22-05c2-4974-9d0f-1b9154bb872e)

Jokes aside, thank you for the suggestion - I'll implement this immediately.

**Comment by Richy-Z** [comment]
As I expected, the GLFW library is excluded from Minecraft servers for obvious reasons. This is an example of the crash log produced using Forge 1.20.2 with Architectury API 10.1.20 (for future reference when this issue is closed):
```log
[19:58:52] [modloading-worker-0/INFO] [ThreatenGL/]: Hark! This numelon mod doth seem to toil as expected! O, dear sire, rejoice! How tremendous!
[19:58:52] [modloading-worker-0/INFO] [ThreatenGL/]: ThreatenGL will threaten Minecraft soon...
[19:58:52] [modloading-worker-0/ERROR] [ne.mi.fm.ja.FMLModContainer/LOADING]: Failed to create mod instance. ModID: threatengl, class com.numelon.threatengl.forge.ThreatenGLForge
java.lang.NoClassDefFoundError: org/lwjgl/glfw/GLFW
        at com.numelon.threatengl.ThreatenGL.init(ThreatenGL.java:18) ~[threatengl-forge-1.2.1-beta.3.jar%2397!/:?] {re:classloading}
        at com.numelon.threatengl.forge.ThreatenGLForge.<init>(ThreatenGLForge.java:16) ~[threatengl-forge-1.2.1-beta.3.jar%2397!/:?] {re:classloading}
        at jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?] {}
        at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[?:?] {}
        at java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[?:?] {}
        at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:70) ~[javafmllanguage-1.20.2-48.1.0.jar%23101!/:?] {}
        at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$10(ModContainer.java:123) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
        at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[?:?] {}
        at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[?:?] {}
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387) ~[?:?] {}
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312) ~[?:?] {}
        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843) ~[?:?] {}
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808) ~[?:?] {}
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188) ~[?:?] {}
Caused by: java.lang.ClassNotFoundException: org.lwjgl.glfw.GLFW
        at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
        at net.minecraftforge.securemodules.SecureModuleClassLoader.loadClass(SecureModuleClassLoader.java:392) ~[securemodules-2.2.3.jar:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
        at net.minecraftforge.securemodules.SecureModuleClassLoader.loadClass(SecureModuleClassLoader.java:392) ~[securemodules-2.2.3.jar:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
        ... 14 more
[19:58:52] [modloading-worker-0/INFO] [ne.mi.co.ForgeMod/FORGEMOD]: Forge mod loading, version 48.1.0, for MC 1.20.2 with MCP 20230921.100330
[19:58:52] [modloading-worker-0/INFO] [ne.mi.co.MinecraftForge/FORGE]: MinecraftForge v48.1.0 Initialized
[19:58:52] [modloading-worker-0/INFO] [de.ar.ne.fo.NetworkManagerImpl/]: Registering C2S receiver with id architectury:sync_ids
[19:58:52] [main/FATAL] [ne.mi.fm.ModLoader/LOADING]: Failed to complete lifecycle event CONSTRUCT, 1 errors found
[19:58:52] [main/ERROR] [minecraft/CrashReport]: Negative index in crash report handler (15/17)
[19:58:52] [main/FATAL] [ne.mi.co.ForgeMod/]: Preparing crash report with UUID 07382bf5-26ed-49c5-8d16-d742a932d480
[19:58:52] [main/FATAL] [ne.mi.se.lo.ServerModLoader/]: Crash report saved to .\crash-reports\crash-2024-05-19_19.58.52-fml.txt
[19:58:52] [main/FATAL] [ne.mi.co.ForgeMod/]: Preparing crash report with UUID 9a222ad9-c05d-4816-9121-6ac72182dd65
---- Minecraft Crash Report ----
// Surprise! Haha. Well, this is awkward.

Time: 2024-05-19 19:58:52
Description: Mod loading error has occurred

java.lang.Exception: Mod Loading has failed
        at net.minecraftforge.logging.CrashReportExtender.dumpModLoadingCrashReport(CrashReportExtender.java:48) ~[forge-1.20.2-48.1.0-universal.jar%23104!/:?] {re:classloading}
        at net.minecraftforge.server.loading.ServerModLoader.load(ServerModLoader.java:37) ~[forge-1.20.2-48.1.0-universal.jar%23104!/:?] {re:classloading}
        at net.minecraft.server.Main.main(Main.java:121) ~[server-1.20.2-20230921.100330-srg.jar%2399!/:?] {re:classloading}
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
        at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
        at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:98) ~[fmlloader-1.20.2-48.1.0.jar%2365!/:?] {}
        at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.lambda$makeService$0(CommonLaunchHandler.java:82) ~[fmlloader-1.20.2-48.1.0.jar%2365!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:17) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:40) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:58) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.Launcher.run(Launcher.java:96) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:13) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:10) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at net.minecraftforge.bootstrap.BootstrapLauncher.main(BootstrapLauncher.java:126) ~[bootstrap-1.2.0.jar:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: main
Suspected Mods: NONE
Stacktrace:
        at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?] {}
-- MOD threatengl --
Details:
        Caused by 0: java.lang.NoClassDefFoundError: org/lwjgl/glfw/GLFW
                at com.numelon.threatengl.ThreatenGL.init(ThreatenGL.java:18) ~[threatengl-forge-1.2.1-beta.3.jar%2397!/:?] {re:classloading}
                at com.numelon.threatengl.forge.ThreatenGLForge.<init>(ThreatenGLForge.java:16) ~[threatengl-forge-1.2.1-beta.3.jar%2397!/:?] {re:classloading}
                at jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?] {}
                at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[?:?] {}
                at java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[?:?] {}
                at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:70) ~[javafmllanguage-1.20.2-48.1.0.jar%23101!/:?] {}
                at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$10(ModContainer.java:123) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
                at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[?:?] {}
                at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[?:?] {}
                at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387) ~[?:?] {}
                at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312) ~[?:?] {}
                at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843) ~[?:?] {}
                at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808) ~[?:?] {}
                at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188) ~[?:?] {}

        Mod File: /C:/Users/Richy/Desktop/forge server/mods/threatengl-forge-1.2.1-beta.3.jar
        Failure message: ThreatenGL (threatengl) has failed to load correctly
                java.lang.NoClassDefFoundError: org/lwjgl/glfw/GLFW
        Mod Version: 1.2.1-beta.3
        Mod Issue URL: https://github.com/Numelon-Softworks/ThreatenGL/issues
        Exception message: java.lang.ClassNotFoundException: org.lwjgl.glfw.GLFW
Stacktrace:
        at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
        at net.minecraftforge.securemodules.SecureModuleClassLoader.loadClass(SecureModuleClassLoader.java:392) ~[securemodules-2.2.3.jar:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
        at net.minecraftforge.securemodules.SecureModuleClassLoader.loadClass(SecureModuleClassLoader.java:392) ~[securemodules-2.2.3.jar:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?] {}
        at com.numelon.threatengl.ThreatenGL.init(ThreatenGL.java:18) ~[threatengl-forge-1.2.1-beta.3.jar%2397!/:?] {re:classloading}
        at com.numelon.threatengl.forge.ThreatenGLForge.<init>(ThreatenGLForge.java:16) ~[threatengl-forge-1.2.1-beta.3.jar%2397!/:?] {re:classloading}
        at jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?] {}
        at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[?:?] {}
        at java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[?:?] {}
        at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:70) ~[javafmllanguage-1.20.2-48.1.0.jar%23101!/:?] {}
        at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$10(ModContainer.java:123) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
        at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[?:?] {}
        at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[?:?] {}
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387) ~[?:?] {}
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312) ~[?:?] {}
        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843) ~[?:?] {}
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808) ~[?:?] {}
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188) ~[?:?] {}


-- System Details --
Details:
        Minecraft Version: 1.20.2
        Minecraft Version ID: 1.20.2
        Operating System: Windows 11 (amd64) version 10.0
        Java Version: 21.0.2, Eclipse Adoptium
        Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Eclipse Adoptium
        Memory: 204606264 bytes (195 MiB) / 377487360 bytes (360 MiB) up to 17129537536 bytes (16336 MiB)
        CPUs: 32
        Processor Vendor: GenuineIntel
        Processor Name: Intel(R) Core(TM) i9-14900KF
        Identifier: Intel64 Family 6 Model 183 Stepping 1
        Microarchitecture: unknown
        Frequency (GHz): 3.19
        Number of physical packages: 1
        Number of physical CPUs: 24
        Number of logical CPUs: 32
        Graphics card #0 name: AMD Radeon RX 6900 XT
        Graphics card #0 vendor: Advanced Micro Devices, Inc. (0x1002)
        Graphics card #0 VRAM (MB): 4095.00
        Graphics card #0 deviceId: 0x73af
        Graphics card #0 versionInfo: DriverVersion=31.0.24031.5001
        Memory slot #0 capacity (MB): 16384.00
        Memory slot #0 clockSpeed (GHz): 6.60
        Memory slot #0 type: Unknown
        Memory slot #1 capacity (MB): 16384.00
        Memory slot #1 clockSpeed (GHz): 6.60
        Memory slot #1 type: Unknown
        Memory slot #2 capacity (MB): 16384.00
        Memory slot #2 clockSpeed (GHz): 6.60
        Memory slot #2 type: Unknown
        Memory slot #3 capacity (MB): 16384.00
        Memory slot #3 clockSpeed (GHz): 6.60
        Memory slot #3 type: Unknown
        Virtual memory max (MB): 69412.09
        Virtual memory used (MB): 22063.11
        Swap memory total (MB): 4096.00
        Swap memory used (MB): 0.00
        JVM Flags: 0 total;
        ModLauncher: 10.1.1
        ModLauncher launch target: forge_server
        ModLauncher naming: srg
        ModLauncher services:
                mixin-0.8.5.jar mixin PLUGINSERVICE
                eventbus-6.2.0.jar eventbus PLUGINSERVICE
                fmlloader-1.20.2-48.1.0.jar slf4jfixer PLUGINSERVICE
                fmlloader-1.20.2-48.1.0.jar object_holder_definalize PLUGINSERVICE
                fmlloader-1.20.2-48.1.0.jar runtime_enum_extender PLUGINSERVICE
                fmlloader-1.20.2-48.1.0.jar capability_token_subclass PLUGINSERVICE
                accesstransformers-8.1.1.jar accesstransformer PLUGINSERVICE
                fmlloader-1.20.2-48.1.0.jar runtimedistcleaner PLUGINSERVICE
                modlauncher-10.1.1.jar mixin TRANSFORMATIONSERVICE
                modlauncher-10.1.1.jar fml TRANSFORMATIONSERVICE
        FML Language Providers:
                minecraft@1.0
                lowcodefml@48
                javafml@null
        Mod List:
                server-1.20.2-20230921.100330-srg.jar             |Minecraft                     |minecraft
        |1.20.2              |COMMON_SET|Manifest: NOSIGNATURE
                forge-1.20.2-48.1.0-universal.jar                 |Forge                         |forge
        |48.1.0              |COMMON_SET|Manifest: 84:ce:76:e8:45:35:e4:0e:63:86:df:47:59:80:0f:67:6c:c1:5f:6e:5f:4d:b3:54:47:1a:9f:7f:ed:5e:f2:90
                architectury.jar                                  |Architectury                  |architectury
        |10.1.20             |COMMON_SET|Manifest: NOSIGNATURE
                threatengl-forge-1.2.1-beta.3.jar                 |ThreatenGL                    |threatengl
        |1.2.1-beta.3        |ERROR     |Manifest: NOSIGNATURE
        Crash Report UUID: 9a222ad9-c05d-4816-9121-6ac72182dd65
        FML: 48.1
        Forge: net.minecraftforge:48.1.0[19:58:52] [main/ERROR] [minecraft/Main]: Failed to start the minecraft server
net.minecraftforge.fml.LoadingFailedException: Loading errors encountered: [
        ThreatenGL (threatengl) has failed to load correctly
§7java.lang.NoClassDefFoundError: org/lwjgl/glfw/GLFW
]
        at net.minecraftforge.fml.ModLoader.waitForTransition(ModLoader.java:246) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
        at net.minecraftforge.fml.ModLoader.lambda$dispatchAndHandleError$20(ModLoader.java:210) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
        at java.util.Optional.ifPresent(Optional.java:178) ~[?:?] {}
        at net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:210) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
        at net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$13(ModLoader.java:183) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
        at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?] {}
        at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:183) ~[fmlcore-1.20.2-48.1.0.jar%23100!/:?] {}
        at net.minecraftforge.server.loading.ServerModLoader.load(ServerModLoader.java:30) ~[forge-1.20.2-48.1.0-universal.jar%23104!/:?] {re:classloading}
        at net.minecraft.server.Main.main(Main.java:121) ~[server-1.20.2-20230921.100330-srg.jar%2399!/:?] {re:classloading}
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
        at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
        at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:98) ~[fmlloader-1.20.2-48.1.0.jar%2365!/:?] {}
        at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.lambda$makeService$0(CommonLaunchHandler.java:82) ~[fmlloader-1.20.2-48.1.0.jar%2365!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:17) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:40) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:58) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.Launcher.run(Launcher.java:96) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:13) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:10) ~[modlauncher-10.1.1.jar%2354!/:?] {}
        at net.minecraftforge.bootstrap.BootstrapLauncher.main(BootstrapLauncher.java:126) ~[bootstrap-1.2.0.jar:?] {}
```

**Comment by Richy-Z** [comment]
This issue has been resolved in ThreatenGL 1.3.1-beta.1, which will be available on Modrinth in a few minutes. ThreatenGL 1.3.0-* will not be uploaded to Modrinth, in favour of this newer version.

Thank you for this suggestion! 🤗
